STYLEGUIDE_CONVENTION.md

```diff
@@ -0,0 +1,199 @@
+  <div>{hello}</div>
+);
+
+// good
+function Listing({ hello }) {
+  return <div>{hello}</div>;
+}
+```
+
+### Extract Login from jsx
+
+```JSX
+//bad
+function FooComponent({ onClick }) {
+ return <button onClick={()=>{
+   ...
+   onClick();
+ }} />
+}
+
+//good
+function FooComponent({ onClick }) {
+ const handleButtonClick = () => {
+  ...
+  onClick();
+ }
+
+ return <button onClick={handleButtonClick} />
+}
+```
+
+### Props Naming
+
+Avoid using DOM component prop names for different purposes.
+
+> Why? People expect props like `style` and `className` to mean one specific thing.
+> Varying this API for a subset of your app makes the code less readable and less maintainable, and may cause bugs.
+
+```JSX
+// bad
+<ExampleComponent style="fancy" />
+
+// bad
+<ExampleComponent className="fancy" />
+
+// good
+<ExampleComponent variant="fancy" />
+```
+
+### Props event naming
+
+```JSX
+// bad
+function ExampleComponent({
+  click,
+  submit,
+  ...
+}) {
+  return ...
+}
+
+// bad
+function ExampleComponent({
+  handleClick,
+  handleSubmit,
+  ...
+}) {
+  return ...
+}
+
+// good
+function ExampleComponent({
+  onClick,
+  onSubmit,
+  ...
+}) {
+  return ...
+}
+
+```
+
+## TypeScript
+
+### Handler event naming
+
+The naming of the handler method contains three parts.
+
+_handle_ _name of element_ _name of event_
+
+```typescript
+// bad
+const buttonclickHandler = (event) => ...
+
+// bad
+const handleClick = (event) => ...
+
+// good
+const handleButtonClick = (event) => ...
+
+// else good
+const handleFormSubmit = (event) => ...
+```
+
+### Variable Naming
+
+boolean variables staring with prefix `is`, `should`, `has`, `can`, `did`, `will`.
+
+### Function Naming
+
+A function or method that returns something must start with a prefix get
+
+```typescript
+function getNumber7() {
+  return 7;
+}
+```
+
+## CSS
+
+[BEM](https://en.bem.info/methodology/quick-start/) is a front-end naming method for organizing and naming CSS classes.
+The Block, Element, Modifier methodology is a popular naming convention for class names in HTML and CSS.
+It helps to write clean CSS by following some simple rules.
+
+There are three main parts of BEM.
+
+1. `Block` which holds everything (elements) inside and acts as a scope.
+2. `Element` which acts as a specific part of the component.
+3. `Modifier` which adds additional styles to a specific element(s).
+
+### Block
+
+A functionally independent page component that can be reused. In HTML, blocks are represented by the `class` attribute.
+
+```HTML
+<!-- `header` block -->
+<header class="header">
+    <!-- Nested `logo` block -->
+    <div class="logo"></div>
+
+    <!-- Nested `search-form` block -->
+    <form class="search-form"></form>
+</header>
+```
+
+### Element
+
+A composite part of a block that can't be used separately from it.
+Identical elements in the same block have the same names. For example, all menu items in the menu block are called `search-form__input`.
```

Comment:
change `all menu items in the menu block` to `all input fields in the search block`